### ISSUE TITLE
Mac viewer mistakenly thinks touchbar is an NDOF device

### ISSUE THREAD

**Issue by nat-goodspeed** [secondlife/viewer#776]
### Environment

Second Life Release 7.1.2.7215179142 (64bit)
Release Notes

CPU: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz (2300 MHz)
Memory: 16384 MB
OS Version: Mac OS X 14.2.1 Darwin 23.2.0 Darwin Kernel Version 23.2.0: Wed Nov 15 21:54:10 PST 2023; root:xnu-10002.61.3~2/RELEASE_X86_64 x86_64
Graphics Card Vendor: ATI Technologies Inc.
Graphics Card: AMD Radeon Pro 5500M OpenGL Engine

OpenGL Version: 4.1 ATI-5.2.4

Window size: 1736x1062
Font Size Adjustment: 96pt
UI Scaling: 1
Draw distance: 64m
Bandwidth: 3000kbit/s
LOD factor: 1.125
Render quality: 0
Advanced Lighting Model: Enabled
Texture memory: 4080MB
Disk cache: Max size 1638.4 MB (89.5% used)
HiDPI display mode: 0

J2C Decoder Version: KDU v7.10.4
Audio Driver Version: FMOD Studio 2.02.13
Dullahan: 1.14.0.202310131309
  CEF: 118.4.1+g3dd6078+chromium-118.0.5993.54
  Chromium: 118.0.5993.54
LibVLC Version: 3.0.16
Voice Server Version: Not Connected
February 08 2024 18:55:26

### Description

Sometimes, on logging into Second Life, my avatar immediately starts flying upwards in a tight spiral, and no UI controls work. I cannot use the viewer at all. The shutdown button works, but very little else.

The log file doesn't show anything peculiar. But when I run it from a console, I see relevant output that apparently doesn't make it into the log file:

```
libndofdev: initializing...

HIDCreateOpenDeviceInterface: Failed to open inDevice->interface via open. #-536870174( 0xffffffffe00002e2 )
hu_BuildDevice: HIDCreateOpenDeviceInterface failed. #-536870174( 0xffffffffe00002e2 )
HIDCreateOpenDeviceInterface: Failed to open inDevice->interface via open. #-536870174( 0xffffffffe00002e2 )
hu_BuildDevice: HIDCreateOpenDeviceInterface failed. #-536870174( 0xffffffffe00002e2 )
HIDCreateOpenDeviceInterface: Failed to open inDevice->interface via open. #-536870174( 0xffffffffe00002e2 )
hu_BuildDevice: HIDCreateOpenDeviceInterface failed. #-536870174( 0xffffffffe00002e2 )
HIDCreateOpenDeviceInterface: Failed to open inDevice->interface via open. #-536870174( 0xffffffffe00002e2 )
hu_BuildDevice: HIDCreateOpenDeviceInterface failed. #-536870174( 0xffffffffe00002e2 )hu_GetCollectionElements: CFDictionaryGetValue error when creating CFTypeRef for kIOHIDElementKey.libndofdev: starting runloop...
2024-02-09T02:51:07Z INFO #Joystick# newview/llviewerjoystick.cpp(382) LLViewerJoystick::init : Failed to gather devices from window. Falling back to ndof's init
Malformed device: found collection at end of element chain.libndofdev: using valid HID device:
libndofdev: manufacturer=Apple Inc.; product=Touch Bar Display; axes_count=6; btn_count=22; min=-3000; max=3000; absolute=1; valid=1; private_data=0x7f95b4bffe70;
    curr_loc_id=FFFFFFFF80600000
    scales:  [0.183111 47.244095 0.183111 47.244095 0.183111 47.244095 ]
    offsets: [-3000.000000 -3000.000000 -3000.000000 -3000.000000 -3000.000000 -3000.000000 ]
2024-02-09T02:51:07Z INFO #Joystick# newview/llviewerjoystick.cpp(432) LLViewerJoystick::init : ndof: mDriverState=1; mNdofDev=NDOF_Device{ axes [0, 0, 0, 0, 0, 0], buttons [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], range -3000:3000, absolute 1, valid 1, manufacturer 'Apple Inc.', product 'Touch Bar Display', private 0x7f95b4bffe70 }; libinit=1
```

### Reproduction steps

Different viewer builds are susceptible to this. I ran the above using the current release viewer just to rule out instability in an upcoming project or dev build. Curiously, there are times when the viewer behaves itself on startup, but other times when it consistently fails the same way.

In short: I don't know how to reproduce it. Run the viewer on an affected Mac.

**Comment by akleshchev** [comment]
This is fixed in DRTVWR-591 in scope of SL-13610 and SL-18830

https://github.com/secondlife/jira-archive-internal/issues/69086

**Comment by kylelinden** [comment]
Closing duplicate
